### PR TITLE
[DSv4] Overlay HCA compressor state onto CSA's NOPE cache

### PR DESCRIPTION
**Pull request by gxd3**
[DSv4] Overlay HCA compressor state onto CSA's NOPE cache (instead of HCA's kv cache)
fix the TODO in https://screenshot-v2.corp.google.com/7wzqnyBVfpsjHfp

With this change, HCA compressor state's block_size change from 2 -> 32; so that the block_table for HCA compressor state become 16x smaller, avoid the SMEM OOM in https://b.corp.google.com/issues/541619368.

Verified with MMLU & MMLU Pro for V4Flash:
https://paste.googleplex.com/6250520430903296
https://paste.googleplex.com/4908257251557376

also piggyback 2 additional small changes:
* let compressor_forward take 1D block_table instead of 2D, so that caller don't need to reshape (tpu-runner's attn-metadata contains 1D block table)
* small perf improvement in mla_swa.py + mla.py

### CODE REVIEW

**Review by a1yssan13 — APPROVED** [review]
thanks guangxiang!

File: tpu_inference/kernels/experimental/deepseek_v4/core_attention/mla_swa.py [hunk 1]
```diff
@@ -571,16 +571,6 @@ def load_bkv(bkv_sem_idx, bkv_idx, start_offset):
         # hi byte). `pltpu.bitcast` fuses consecutive sublanes little-endian, so the
         # reconstruction is a plain bitcast + reshape.
         bkv = pltpu.bitcast(bkv_u8, jnp.bfloat16).reshape(bkv_sz, head_dim)
-
```
Comment (inucool12) [inline tpu_inference/kernels/experimental/deepseek_v4/core_attention/mla_swa.py:574]: Since SWA caches still overlay the CSA NoPE array (which is shared with CSA/HCA states), isn't  there any risk of removing NaN mask here ?

Comment (gxd3) [reply tpu_inference/kernels/experimental/deepseek_v4/core_attention/mla_swa.py:574]: The reason for no risk:
in fetch_bkv(), we only DMA the valid kv (must not NaN) into `bkv_x2_ref` (not DMA the whole page, but DMA the valid tokens' kv)

during init: we zero the buffer: https://screenshot-v2.corp.google.com/6YvRhHWwNbxEFa6 at the begining.

therefore, we never bring NaN into bkv_x2_ref ever; so it's safe.
